prime-router/src/main/kotlin/tokens/Scope.kt

```diff
@@ -16,26 +15,71 @@ class Scope {
 
         /**
          * Scope that represents prime admin access.
+         * Pattern is <organization>.<receiver-or-sender>.<DetailedScope>
+         * The first value must always be an organization name, or wildcard
+         * The idea is to leave the last field loose; it could be a role or a permission
+         * Examples:
+         * md-phd.*.user -- user level access
+         * md-phd.default.report -- able to submit reports only
+         * md-phd.*.admin
+         * *.*.primeadmin
          */
-        val primeAdminScope = "primeadmin.all.all"
+        val primeAdminScope = "*.*.primeadmin"
 
-        enum class DetailedScope {
-            report, // ability to submit a report
-            all, // all abilities
+        /**
+         * Constraint set of allowed values for the last part of a scope string.
+         * [strRep] is the string representation of the DetailedScope.
+         */
+        enum class DetailedScope(val strRep: String) {
+            Report("report"), // ability to submit a report
+            Admin("admin"), // local org administrator role
+            PrimeAdmin("primeadmin"), // overall administrator role
+            User("user"); // user role
+
+            companion object {
+                /**
+                 * Convert a string representation of a DetailedScope to a DetailedScope obj.
+                 * @return the matching DetailedScope, or null if no such DetailedScope exists.
+                 */
+                fun toDetailedScope(strRep: String): DetailedScope? {
+                    return values().find { it.strRep == strRep }
+                }
+            }
+        }
+
+        /**
+         * Return true if scope is the PrimeAdmin scope
+         */
+        fun isPrimeAdmin(scope: String?): Boolean {
+            return scope == primeAdminScope
         }
 
+        /**
+         * @return true if this [scope] syntax is correct.  Otherwise false.
```

Comment:
Is a wildcard in the third position valid syntax?  IE *.*.* ?

Reply:
No, third position is currently defined by an enum set of limited values.   I thought about having *.*.* be the primeadmim capability, but decided it would be better to have the word 'primeadmin' specifically in the string, for clarity.